### DISCUSSION TITLE
Why does MAD-X link against X11 libraries? I don't see any documented way to plot to the screen.

### DISCUSSION THREAD

**Discussion by jsberg-bnl** [MethodicalAcceleratorDesign/MAD-X#1046] [Q&A]
Is there a way to plot to the screen in MAD-X? I don't see any indication of a parameter for that in the documentation. MAD-X does link against the X11 libraries, but I'm not clear on why.

**Comment by rdemaria** [comment]
By dafault, when you plot on linux in interactive mode, e.g.

```
...
    plot,table=my_target1,title="Target orbit 1 x...",noline=true,vmin=-0.012,vmax=0.012,haxis=s,vaxis=x;
...
```
You get the following prompt

``` 
 GX (X11 based) plot package initialization
 Do you want to plot on your terminal ? (<CR> = yes>:
 specify bounding box size (default: 27(x) by 19(y) cm)?
 (<CR>=no):
```
then an X11 window opens.

In my opinion plotting in madx should be deprecated because it is expensive to maintain (in particular building with X11 dependency is difficult) and there are many other easy ways to make beautiful plots...

**Comment by kyrsjo** [comment]
If deprecating all plotting, there should really be a "standard" and documented way of achieving the same... Prefferably an official python etc. package :)

Still, the PostScript plots is a quite nice and easy way to debug things, where you get a bunch of plots one after the other in the same file...
